src/app/modules/auth/sign-up/sign-up.component.ts

```diff
@@ -68,27 +75,24 @@ export class AuthSignUpComponent implements OnInit {
         this.showAlert = false;
 
         // Sign up
-        this._authService.signUp(this.signUpForm.value).subscribe(
-            (response) => {
-                // Navigate to the confirmation required page
-                this._router.navigateByUrl('/sign-in');
-            },
-            (response) => {
-                // Re-enable the form
-                this.signUpForm.enable();
+    this._authService.signUp(this.signUpForm.value).subscribe({
+        next: () => this._router.navigateByUrl('/sign-in'),
+        error: () => {
+        this.signUpForm.enable();
```

Comment:
this._authService.signUp(this.signUpForm.value)
.pipe( // pipe for doing stuff with the stream with operators
tap(() => this._router.navigateByUrl('/sign-in'), // tap needed for side effects from observable 
catchError(() => {
// whatever you do in case of error 
})
),
).subscribe(); // to subscribe to the whole observable

Reply:
catchError should not be inside tap